pkg/hubble/metrics/api/api.go

```diff
@@ -146,3 +128,101 @@ var registry = NewRegistry(
 func DefaultRegistry() *Registry {
 	return registry
 }
+
+// FlowFilters is a slice of filters to include or exclude flows.
+type FlowFilters []*pb.FlowFilter
+
+type ContextValues []string
+
+// ContextOptions is a structure used for configuring Hubble metrics context options
+type ContextOptionConfig struct {
+	Name   string        `json:"name,omitempty" yaml:"name,omitempty"`
+	Values ContextValues `json:"values,omitempty" yaml:"values,omitempty"`
+}
+
+type ContextOptionConfigs []*ContextOptionConfig
+
+type MetricConfig struct {
+	Name                 string               `json:"name,omitempty" yaml:"name,omitempty"`
+	ContextOptionConfigs ContextOptionConfigs `json:"contextOptions,omitempty" yaml:"contextOptions,omitempty"`
+	IncludeFilters       FlowFilters          `json:"includeFilters,omitempty" yaml:"includeFilters,omitempty"`
+	ExcludeFilters       FlowFilters          `json:"excludeFilters,omitempty" yaml:"excludeFilters,omitempty"`
+}
+
+// DynamicMetricsConfig represents the structure used for configuring
+// Hubble metrics context options
+type DynamicMetricsConfig struct {
+	Metrics []*MetricConfig `json:"metrics,omitempty" yaml:"metrics,omitempty"`
+}
+
+func (d *DynamicMetricsConfig) GetMetricNames() map[string]struct{} {
+	metrics := make(map[string]struct{})
+	for _, m := range d.Metrics {
+		metrics[m.Name] = struct{}{}
+	}
+	return metrics
+}
+
+func ParseStaticMetricsConfig(enabledMetrics []string) (metricConfigs *DynamicMetricsConfig) {
+	metricConfigs = &DynamicMetricsConfig{}
+	for _, metric := range enabledMetrics {
+		s := strings.SplitN(metric, ":", 2)
+		config := &MetricConfig{
+			Name:                 s[0],
+			IncludeFilters:       FlowFilters{},
+			ExcludeFilters:       FlowFilters{},
+			ContextOptionConfigs: ContextOptionConfigs{},
+		}
+		if len(s) == 2 {
+			config.ContextOptionConfigs = parseOptionConfigs(s[1])
+		}
+		metricConfigs.Metrics = append(metricConfigs.Metrics, config)
+	}
+
+	return
+}
+
+func parseOptionConfigs(s string) (options ContextOptionConfigs) {
+	options = ContextOptionConfigs{}
+
+	for _, option := range strings.Split(s, ";") {
+		if option == "" {
+			continue
+		}
+
+		kv := strings.SplitN(option, "=", 2)
+		ctxOption := &ContextOptionConfig{
+			Name: kv[0],
+		}
+		if len(kv) == 2 {
+			ctxOption.Values = parseOptionValues(kv[1])
+		} else {
+			ctxOption.Values = []string{""}
+		}
+		options = append(options, ctxOption)
+	}
+
+	return
+}
+
+func parseOptionValues(s string) (values ContextValues) {
+	values = ContextValues{}
+
+	if strings.Contains(s, "|") {
+		for _, option := range strings.Split(s, "|") {
+			if option == "" {
+				continue
+			}
+			values = append(values, option)
+		}
+	} else {
+		// temporarily handling comma separated values for labels context
+		for _, option := range strings.Split(s, ",") {
```

Comment:
@chancez  this is a weird case, Hubble uses two kinds of delimiters "|" for the contextOptions and "," for labelcontexts.

Reply:
Yeah that's because we needed another delimiter for nested config within the already nested config.